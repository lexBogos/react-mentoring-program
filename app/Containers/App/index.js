import React from "react";
import Header from "../Header";
import Main from "../Main";
import Footer from "../Footer";
import Modal from "../../components/Modal";
import NoMatch from "../../components/NoMatch";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

const App = () => {
  return (
    <Router>
      <Switch>
        <Route path="/" exact>
          <Header />
          <Main />
          <Modal />
        </Route>
        <Route path="/movie/:id" exact>
          <Header />
          <Main />
          <Modal />
        </Route>
        <Route path="*">
          <NoMatch />
        </Route>
      </Switch>
      <Footer />
    </Router>
  );
};

export default App;
