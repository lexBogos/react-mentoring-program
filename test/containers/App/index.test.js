import React from 'react';
import App from '../../../app/containers/App/index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme,{ shallow } from 'enzyme';
import store from '../../../app/store';
import '@testing-library/jest-dom/extend-expect';

Enzyme.configure({ adapter: new Adapter() });

describe('App component', () => {
        it('render App component toMatchSnapshot', () => {
            expect(shallow(<App store={store}/>).debug()).toMatchSnapshot();
        })
})