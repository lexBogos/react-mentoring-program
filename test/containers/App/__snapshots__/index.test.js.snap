// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render App component toMatchSnapshot 1`] = `
"<BrowserRouter>
  <Switch>
    <Route path=\\"/\\" exact={true}>
      <Component />
      <Main />
      <Component />
    </Route>
    <Route path=\\"/movie/:id\\" exact={true}>
      <Component />
      <Main />
      <Component />
    </Route>
    <Route path=\\"*\\">
      <NoMatch />
    </Route>
  </Switch>
  <Footer />
</BrowserRouter>"
`;
